src/app/(admin)/new_dashboard/guides/components/ReviewList/index.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+import React from "react";
+import { Typography, Box, Rating, Divider,CircularProgress, Alert } from "@mui/material";
+import { Review } from "@/api/types/Review"; 
+import PageContainer from "@/components/PageContainer";
+import DashboardCard from "../../../shared/Card";
+import AddReviewBox from "../AddReviewBox";
+import { useReviews } from "../../hooks";
+
+interface ReviewListProps {
+  guideId: number;
+}
+
+const ReviewList: React.FC<ReviewListProps> = ({ guideId }) => {
+  const {data:response , isError , error , isSuccess  ,isLoading} = useReviews(guideId);
+  let reviews: Review[] = [];
+  if (isLoading) {
+    return <CircularProgress />;
+  }
+
+  if(isError) {
+    return <Alert severity="error">This is an error Alert.</Alert>;
+  }
+
+  if(isSuccess && "data" in response) {
+    reviews = response.data;
+    if(reviews.length === 0) {
+      return (
+        <PageContainer title="Reviews">
+          <DashboardCard title="Reviews">
+            <Box mt={2}>
+              <Typography>No reviews available</Typography>
+              <Divider sx={{ mt: 2, mb: 2 }} />
+              <AddReviewBox guideId={guideId} />
+            </Box>
+          </DashboardCard>
+        </PageContainer>
+      )
+    }
```

Comment:
Properly type the error in the Alert.

Also try a different approach instead of the let..

Reply:
As I said before I am open to suggestions, lets think about it together